services/bls_aggregation/blsagg.go

```diff
@@ -234,21 +194,109 @@ func NewBlsAggregatorService(
 	logger logging.Logger,
 ) *BlsAggregatorService {
 	return &BlsAggregatorService{
-		aggregatedResponsesC: make(chan BlsAggregationServiceResponse),
-		signedTaskRespsCs:    make(map[types.TaskIndex]chan types.SignedTaskResponseDigest),
-		taskChansMutex:       sync.RWMutex{},
-		avsRegistryService:   avsRegistryService,
-		logger:               logger,
-		hashFunction:         hashFunction,
+		avsRegistryService: avsRegistryService,
+		logger:             logger,
+		hashFunction:       hashFunction,
 	}
 }
 
-func (a *BlsAggregatorService) GetResponseChannel() <-chan BlsAggregationServiceResponse {
-	return a.aggregatedResponsesC
+// The service handler is a structure used to use the service without the complexity of it.
+type ServiceHandler struct {
+	//This channels are used to send messages (requests) to the service.
+	TaskInitC         chan InitializeTaskRequest
+	processSignatureC chan ProcessSignatureRequest
 }
 
-// InitializeNewTask creates a new task goroutine meant to process new signed task responses for that task
-// (that are sent via ProcessNewSignature) and adds a channel to a.taskChans to send the signed task responses to it.
+type InitializeTaskRequest struct {
```

Comment:
Let's make this private:

```suggestion
type initializeTaskRequest struct {
```

Reply:
Done in [8fb5557](https://github.com/Layr-Labs/eigensdk-go/pull/578/commits/8fb555786f57f420f2ac34d0b8fd17024b1d7a46)